code/FsComposite/Composite.Core/Processing.fs

```diff
@@ -4,22 +4,51 @@ open Composite.Core.Composite
 
 module Processing =
     
+    let rec applyScnToElement scn acc element =
+        match scn with
+        | [] -> acc
+        | f :: scn_tail -> match f element with
+                           | Cons(Nil, Nil) -> applyScnToElement scn_tail acc element
+                           | x -> applyScnToElement scn_tail (LazyList.append acc x) element

```

Comment:
The thing we aim to do here is redesigning `cata` around a `map` which would pass recursively through the leaves of a `Composite` with a function like this. 

The function, however, should be based on stuff like `find_3_and_transform` only with lists in place of tuples and a single element in place of the sequence.

Then the function cata does the following:

* it walks about the `Composite` and runs each leaf through a sequence of scenario steps one step at a time;
* if the step yields a result (each step returns a tuple of accumulator list and a result list) the step is excluded from the sequence used on further leaves and the result is sent to out sequence;
* if the step yields no result, the accumulator is passed to the same step when the next leaf is processed.